maestro-app/src/main/java/bio/overture/maestro/app/infra/adapter/outbound/indexing/elasticsearch/FileCentricElasticSearchAdapter.java

```diff
@@ -149,16 +151,59 @@ private IndexResult bulkIndexFiles(List<FileCentricDocument> filesList) {
     private IndexResult bulkUpsertFileRepositories(List<FileCentricDocument> filesList) {
         log.trace("in bulkUpsertFileRepositories, filesList count : {} ", filesList.size());
         val size = this.documentsPerBulkRequest;
-        partitionList(filesList, size)
-            .forEach((partNum, listPart)-> {
-                log.trace("bulkUpsertFileRepositories, sending part#: {}, hash: {} for filesList hash: {} ", partNum,
-                    Objects.hashCode(listPart), Objects.hashCode(filesList));
-                this.customElasticSearchRestAdapter.bulkUpdateRequest(listPart.stream()
-                    .map(this::mapFileToUpsertRepositoryQuery)
-                    .collect(Collectors.toList())
-                );
+        val failures = partitionList(filesList, size)
+            .entrySet()
+            .stream()
+            .map((entry)-> tryBulkUpsertRequestForPart(filesList, entry))
+            .flatMap(Set::stream)
+            .collect(Collectors.toUnmodifiableSet());
+
+        val fails = FailureData.builder()
+            .failingIds(Map.of(ANALYSIS_ID, failures))
+            .build();
+
+        return IndexResult.builder()
+            .failureData(fails)
+            .successful(failures.isEmpty())
+            .build();
+    }
+
+    @NotNull
+    private Set<String> tryBulkUpsertRequestForPart(List<FileCentricDocument> filesList,
+                                                    Map.Entry<Integer, List<FileCentricDocument>> entry) {
+        val partNum = entry.getKey();
+        val listPart = entry.getValue();
+        val listParthHash = Objects.hashCode(listPart);
+        val fileListHash =  Objects.hashCode(filesList);
+        val retryConfig = RetryConfig.custom()
+            .maxAttempts(3)
+            .retryExceptions(IOException.class)
+            .waitDuration(Duration.ofMillis(500))
+            .build();
+        val retry = Retry.of("bulkUpsertFileRepositoriesRetry", retryConfig);
+        val decorated = Retry.decorateCheckedSupplier(retry, () -> {
```

Comment:
This is pretty cool @blabadi

Reply:
yep this lib (https://github.com/resilience4j/resilience4j) is really nice